src/frame/sdk/tests/UserApi.test.ts

```diff
@@ -39,6 +39,36 @@ describe("UserApi", () => {
         });
     });
 
+    describe("rotateUserMasterKey", () => {
+        it("sends message to worker and takes result and passes it to call user key update endpoint", (done) => {
+            ApiState.setCurrentUser(TestUtils.getFullUser());
+            spyOn(UserApiEndpoints, "callUserKeyUpdateApi").and.returnValue(Future.of("user key update result"));
+            spyOn(WorkerMediator, "sendMessage").and.returnValue(
+                Future.of({
+                    message: {newEncryptedPrivateUserKey: "newEncryptedPrivateUserKey", augmentationFactor: "augmentationFactor"},
+                })
+            );
+            spyOn(ApiState, "setEncryptedPrivateUserKey");
```

Comment:
No need to change this, but because this call doesn't have any side effects that we don't want (like making requests to the API or similar) mocking it out isn't really necessary. Instead of doing a mock and then asserting that it was called below, you could just not mock things and assert that the result of `ApiState.encryptedUserKey` is what you'd expect. The goal here is mostly to avoid mocking out everything and only mock out the parts that will cause a bunch of problems when testing.